pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -924,6 +924,22 @@ func (a *ArtifactLocation) GetType() ArtifactLocationType {
 
 }
 
+func (a *ArtifactLocation) GetKey() string  {
+	if a.S3 != nil {
+		return a.S3.Key
+	}
+
+	if a.OSS != nil {
+		return a.OSS.Key
+	}
+
+	if a.GCS != nil {
+		return a.GCS.Key
+	}
+
+	return ""
```

Comment:
Should this return an error instead of `""`?  Went back and forth on this a bit since only a subset of `ArtifactLocation`s support outputs and therefore keys.

Alternatively, I could do this logic in the artifact server itself.

Reply:
empty is fine